src/app-layer-htp-range.h

```diff
@@ -0,0 +1,81 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#ifndef __APP_LAYER_HTP_RANGE_H__
+#define __APP_LAYER_HTP_RANGE_H__
+
+void HttpRangeContainersInit(void);
+void HttpRangeContainersDestroy(void);
+uint32_t HttpRangeContainersTimeoutHash(struct timeval *ts);
+
+void *HttpRangeContainerUrlGet(const uint8_t *key, size_t keylen, struct timeval *ts);
+
+// linked list of ranges : buffer with offset
```

Comment:
Ok, I hope I got it right...

I do some
```
RB_FOREACH(...) {
  if (condition) {
    break;
  }
  // do stuff
  RB_REMOVE(...);
}
```
This felt better with a linked list... (I am afraid this is O(n ln n) instead of O(n) )
Is there a better way to cut the red and black tree off some of its smallest elements ?